jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ProcessorMXBeanImpl.java

```diff
@@ -117,4 +117,30 @@ public int getNumberBoundCPUs() {
 	public int getNumberTargetCPUs() {
 		return getNumberCPUsImpl(TARGET);
 	}
+	
+	/**
+	 * @param number
+	 * 			The number of CPUs to specify the process to use. The process
+	 * 			will behave as if <code>number</code> CPUs are available.
+	 * 
+	 * @see #setNumberActiveCPUs(int)
+	 */
+	native void setNumberActiveCPUsImpl(int number);
+
+	/**
+	 * {@inheritDoc}
+	 */
+		@Override
+	public void setNumberActiveCPUs(int number) throws IllegalArgumentException {
+		SecurityManager security = System.getSecurityManager();
+		if (security != null) {
+			security.checkPermission(ManagementPermissionHelper.MPCONTROL);
+		}
+
+		if (0 > number) {
```

Comment:
Double checking with HotSpot, I found that they don't throw an error on negative numbers either, and only consider active cpu's with a valid input. Should we duplicate this behavior or throw an error on all invalid inputs?

Reply:
What is the scenario you tested exactly? Does Hotspot allow setting the active cpus via java code?

Reply:
I tested using the `-XX:ActiveProcessorCount` flag - I really should have asked two separate questions:
1. Should the flag have the exact same behaviour as HotSpot?
2. Should `setNumberActiveCPUs` behave the same as the flag? (I should not have assumed the answer to this be yes)

Reply:
Also, I have not found any java API to set active cpus in HotSpot.

Reply:
It makes sense to me to set zero cpus, which removes the active cpu settings. However the doc and the tests should be updated to cover this.

I suppose there is nothing wrong with setting zero cpus via the command line option, it just doesn't do anything. How does HotSpot handle zero?

Reply:
HotSpot's command line option accepts zero and negative numbers without throwing an error. In this case, it seems to have no effect (`Runtime.availableProcessors()` , at least, is not overridden, and I can't imagine what other course of action the JVM could take).

Reply:
From looking at the code and a simple experiment, OpenJ9 won't accept negative numbers because its parsing for an unsigned value. OpenJ9 already accepts zero and does nothing, which matched the Hotspot behavior. I think the OpenJ9 command line operation is fine.

Reply:
Also, I was under the impression that ProcessorMXBean is not properly installed as an MXBean and is essentially unused code - I added back whatever was already in place (before entitled CPUs was removed) with updated comments and function names since this PR is mostly just undoing #2236 removals. 

Should this just be completely scrapped? If not, is there a way for tests to be written for this API?

Reply:
The code can be called directly (in Java 8), as shown by the testing, its just not a proper MXBean. I think updating it is correct. We could expose it as a MXBean in a subsequent change.

Reply:
Sorry, I guess there is no testing for this class, but the code can be called directly by creating an instance of ProcessorMXBeanImpl

Reply:
To be clear, I'm not asking for tests for ProcessorMXBeanImpl in order to merge this.

Reply:
OK, understood. I was confused earlier about how this API was exposed and what the implications of that were - thanks for the clarification